MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```
@@ -940,6 +941,129 @@ private void OnSetAnimationState(SetAnimationState payload)
+                var soundInstance = _soundInstances[id];
+                if (!soundInstance.isPlaying)
+                {
+                    DestroySoundInstance(soundInstance, id);
+                }
+                else
+                {
+                    soundStoppedCheckIndex++;
+                }
+            }
+        }
+
+        private void DestroySoundInstance(AudioSource soundInstance, Guid id)
+        {
+            Component.Destroy(soundInstance);
+            _soundInstances.Remove(id);
+            var index = _unpausedSoundInstances.FindIndex(x => x == id);
+            if (index >= 0)
+            {
+                _unpausedSoundInstances.RemoveAt(index);
+            }
+        }
+        [CommandHandler(typeof(SetSoundState))]
+        private void OnSetSoundState(SetSoundState payload)
+        {
+            var actor = _actorManager.FindActor(payload.ActorId);
+            if (actor != null)
+            {
+                if (payload.SoundCommand == SoundCommand.Start)
+                {
+                    var obj = MREAPI.AppsAPI.AssetCache.GetAsset(payload.SoundAssetId);
+                    var audioClip = MREAPI.AppsAPI.AssetCache.GetAsset(payload.SoundAssetId) as AudioClip;
+                    if (audioClip != null)
+                    {
+                        var soundInstance = actor.gameObject.AddComponent<AudioSource>();
+                        soundInstance.clip = audioClip;
+                        soundInstance.time = payload.StartTimeOffset;
+                        ApplySoundStateOptions(soundInstance, payload.Options);
+                        soundInstance.Play();
+                        _soundInstances.Add(payload.Id, soundInstance);
+                        _unpausedSoundInstances.Add(payload.Id);
+                    }
+                }
+                else
+                {
+                    var soundInstance = _soundInstances[payload.Id];
```

Is there cases where this id coming in the payload could not be in this dictionary?  If so, this should raise a KeyNotFound exception, and should be avoided by using the contains key test around this or the TryGetValue method.

---

fixed everywhere

---
In reply to: [260559537](https://github.com/Microsoft/mixed-reality-extension-unity/pull/35#discussion_r260559537) [](ancestors = 260559537)